### PR TITLE
Remove most `*_kernels.jl` files

### PR DESCRIPTION
**Pull request by jkozdon**
# Description

There is no longer a need to separate the kernel file from rest of the module. Since many of the kernel files only have one or two functions, it might be cleaner if they are removed.

The one file I did not remove was `src/Numerics/DGmethods/DGmodel_kernels.jl` because it is >2000 lines.

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [x] CLIMA developers with relevant expertise have been assigned to review this submission
- [x] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by jkozdon** [comment]
bors try

File: src/Arrays/MPIStateArrays.jl [hunk 1]
```diff
@@ -784,6 +784,46 @@ end
 
 @init tictoc()
 
-include("MPIStateArrays_kernels.jl")
+using Requires
+@init @require CUDAnative = "be33ccc6-a3ff-5ff2-a52e-74243cff1e17" begin
+    using .CUDAnative
```
Comment (vchuravy) [inline src/Arrays/MPIStateArrays.jl:789]: Do we still need this?

**Comment by jkozdon** [comment]
bors try

File: src/Arrays/MPIStateArrays.jl (hunk 1 above)
Comment (jkozdon) [reply src/Arrays/MPIStateArrays.jl]: Not sure. Will try to remove and see.

Comment (jkozdon) [reply src/Arrays/MPIStateArrays.jl]: Fine to remove will fix up the commits and push this out.

**Comment by jkozdon** [comment]
I'm at a loss for why the [mac CI](https://dev.azure.com/climate-machine/CLIMA/_build/results?buildId=3445&view=logs&j=93056758-5bfb-5750-f113-e720ddefdb4c&t=c73b3de0-27be-598b-c922-0c24c64f8585&l=886) is failing on cg. Thoughts?

cc: @leios

**Comment by leios** [comment]
The mac CI seems to be passing now, and I genuinely cannot replicate the error at this point in time. I will look through any other PR's that seem to be failing for this reason to see if I can get more information.

**Comment by jkozdon** [comment]
bors try

**Comment by jkozdon** [comment]
> The mac CI seems to be passing now, and I genuinely cannot replicate the error at this point in time. I will look through any other PR's that seem to be failing for this reason to see if I can get more information.

Just seemed weird that it failed twice for this PR and not the others.

I couldn't reproduce the error locally. We'll see what comes back on the "try" I just did.

**Comment by leios** [comment]
Agreed. I couldn't replicate it either, but @sandreza could and he noticed that one of the arrays were initialized with NaN's already in it, hence #1008 .

**Pull request merged**